spec/models/auth_token_spec.rb

```diff
@@ -2,14 +2,28 @@
   include TokenHelper
 
   describe "#data" do
-    it "returns the data hash when valid" do
-      token = AuthToken.new(encrypt_and_sign_token(data: { a: "b" }))
-      expect(token.data).to eq(a: "b")
+    context "with a SHA1 token" do
+      it "returns the data hash when valid" do
+        token = AuthToken.new(encrypt_and_sign_token(data: { a: "b" }, hash_digest_class: OpenSSL::Digest::SHA1))
+        expect(token.data).to eq(a: "b")
+      end
+
+      it "returns nil after the expiry time" do
+        token = AuthToken.new(encrypt_and_sign_token(expiry: 0, hash_digest_class: OpenSSL::Digest::SHA1))
```

Comment:
These two `returns nil` tests don't give me much confidence as they don't pass the `data: { a: "b" }` part of the hash, so even without an `expiry: 0`, I'd expect `token.data` to be `nil` 🤔 
Could you add the `data` property to explicitly test this?

Reply:
The default data is actually an empty hash, so if you set the expiry to (say) 1000, or remove it entirely, the test will fail - it's only nil if the token is invalid for some reason. BUT! there's actually a nicer way of doing this - I can expand the test suite and move these tests into a test for the #valid? method, which would be a bit more explicit.

Reply:
Have updated the tests, see if the current change seems less worrying.